python/ray/tune/tests/test_trial_runner.py

```diff
@@ -1639,6 +1639,38 @@ def testRestoreMetricsAfterCheckpointing(self):
         self.assertGreater(trials[1].last_result["time_since_restore"], 0)
         self.addCleanup(os.remove, path)
 
+    def testRestoreNewestMetricsAfterCheckpointing(self):
+        ray.init(num_cpus=1, num_gpus=1)
+        runner = TrialRunner()
+        kwargs = {
+            "resources": Resources(cpu=1, gpu=1),
+        }
+        runner.add_trial(Trial("__fake", **kwargs))
+        trials = runner.get_trials()
+
+        runner.step()
+        self.assertEqual(trials[0].status, Trial.RUNNING)
+        self.assertEqual(ray.get(trials[0].runner.set_info.remote(1)), 1)
+        path = runner.trial_executor.save(trials[0])
+        runner.trial_executor.stop_trial(trials[0])
+        kwargs["restore_path"] = os.path.abspath(os.path.join(path, "../.."))
+
+        runner.add_trial(Trial("__fake", **kwargs))
+        trials = runner.get_trials()
+
+        runner.step()
+        self.assertEqual(trials[0].status, Trial.TERMINATED)
+        self.assertEqual(trials[1].status, Trial.RUNNING)
+        runner.step()
```

Comment:
How does this check it restores the newest metrics? I think you need to run multiple times in order to see that you actually checkpointed the "newest" rather than the only one

Reply:
Thanks! I will modify the test case.

Reply:
ping?